### ISSUE TITLE
where can I disable these (daily?) ocsp.*.com requests?

### ISSUE THREAD

**Issue by springy76** [gitextensions/gitextensions#6510]
When I start GitExtensions (by either Windows Explorer extension or direct app open) it takes a minute or two till any screen appears, because while connected by VPN all non-company connections to internet only timeout.

Where can I disable these initial requests to ocsp.usertrust.com and ocsp.comodoca.com which I see in Fiddler?

**Comment by pmiossec** [comment]
I don't recognize the URL but perhaps it is solved by #6287

Could you please test a development build https://github.com/gitextensions/gitextensions/blob/master/README.md#downloads

**Comment by gerhardol** [comment]
GE calls GitHub periodically if you have configured checks in settings.
For Avatars, you do not really need master, just disable avatars in commit info and the grid. (3.1 will add another setting though that makes it more visible.)

**Comment by springy76** [comment]
GE only does this once a day (or anything greater than every 1 hour) and it always does 3 requests, one to ocsp.usertrust.com and two to ocsp.comodoca.com. Avatar requests -- if any -- occur later; the pull dialog opened from Explorer shell extension doesn't show/request avatars, but also is delayed for a minute.

**Comment by illfated** [comment]
According to what I can find with a Google search on those 2 URLs, this is not about avatars. Instead it is for verifying certificates (crypto) via Comodo (https://ssl.comodo.com/):
> OCSP is a newer web service that allows clients to verify if an SSL certificate has been revoked. The older standard, CRL (Certificate Revocation List) required that browsers download the entire list. With OCSP, it is possible to query the status of an individual certificate. The certificate has to have the URL for the respective CRL or OCSP service embedded.

**Comment by RussKie** [comment]
We don't call either of these websites directly from our code

![image](https://user-images.githubusercontent.com/4403806/56863577-65f87400-69fb-11e9-95b4-6da52186920c.png)
![image](https://user-images.githubusercontent.com/4403806/56863581-76105380-69fb-11e9-93da-7a2bf4f49b14.png)

Can you please elaborate what payload these requests carry?

I can speculate it is the Windows Smartscreen is checking the certificate that our binaries are signed with.

**Comment by springy76** [comment]
I enabled SSL decryption in Fiddler but guest what: Today no request occured :-|

I'll keep watching. While connected by VPN I don't start much apps, most of time only GE and VS (for fetching R# license token from company license server), but VS never is delayed.

Luckily starting GE once before opening VPN prevents that delay.

**Comment by RussKie** [comment]
I'm going to close it since I don't believe there is anything we can do.
Please keep us posted if you find something.

**Comment by springy76** [comment]
Today I had them again with Fiddler+SSL-decryption enabled:

```
GET http://ocsp.usertrust.com/MFEwTzBNMEswSTAJBgUrDgMCG...
GET http://crl.usertrust.com/AddTrustExternalCARoot.crl
GET http://ocsp.comodoca.com/MFEwTzBNMEswSTAJBgUrDgMCGg...
GET http://crl.comodoca.com/COMODORSACertificationAuthority.crl
GET http://ocsp.comodoca.com/MFIwUDBOMEwwSjAJBgUrDgMCGg...
GET http://crl.comodoca.com/COMODORSACodeSigningCA.crl
```

Fiddler associated all 5 of them with the process named "GitExtensions" but the User-Agent header says "Microsoft-CryptoAPI/6.1" -- can't tell if it really was in-process or faked by MSE.

**Comment by illfated** [comment]
Comodo used to provide a free personal encryption certificate, as a means to establish a free digital signature ID in Microsoft Outlook for verification and encryption. The free signature ID was supposedly only useful for demonstration purposes in a one-to-one communication situation, but apparently Comodo have since removed the free certification (unless already subscribed). I don't know how much of this is relevant for your issue, just mentioning it in case it could be useful somehow.

**Comment by RussKie** [comment]
@stefan-wenig do you know anything about these calls?

**Comment by mstv** [comment]
Could it be that the Windows library tries to check whether the certificate of an HTTPS is server is valid?

**Comment by illfated** [comment]
> Could it be that the Windows library tries to check whether the certificate of an HTTPS server is valid?

Looks very likely to me, but I don't know for certain.

**Comment by stefan-wenig** [comment]
http://crl.comodoca.com/COMODORSACodeSigningCA.crl is the CRL url in our certificate
http://crl.comodoca.com/COMODORSACertificationAuthority.crl is from the intermediate cert
http://ocsp.comodoca.com is the referenced in both of them
http://crl.usertrust.com might be from a cross-singed intermediate, not sure

So yes, this is caused by code signing. Also it's interesting that it seems that only the .exe file's cert is checked, but not those of your DLLs. 

However, I cannot say what triggers these checks. Windows should only do SmartScreen and UAC checks upon installation (i.e. once). MSE sounds like a plausible candidate. Do you use AppLocker or something similar? Maybe check the logs of whatever security software you're using?

(In .NET 3.0 there was also the issue of CAS Publisher Evidence generation, which could create this exact problem. However, in .NET 4.0 this should be gone except for .NET-based NT services. Google generatePublisherEvidence for more info.)

In short, while your issue is _related_ to GitExtensions' code signing signature, it's not _caused_ by GitExtensionsor or its signature.

**Comment by RussKie** [comment]
Thank you Stefan 👍

**Comment by springy76** [comment]
I still wonder why the request get associated (by Fiddler) with the process named "GitExtensions". Does MSE (I don't use anything other) inject DLLs which act on behalf of the injected process?

**Comment by roamergu** [comment]
hey i  got same problem in our PC, i take windows server 2012R2 to our DC, and windows 10 in our domain, now if windows 10 quit AD, git-bash work fine, if add in AD, then git-bash access comodoca.com and keep try, i m in china and these url about comodo can't access.

**Comment by pmiossec** [comment]
Sectigo/comodoca is the ssl certificat used by gravatar web site.
So maybe you have just to disable this feature in Git Extensions as advised in the first messages....

**Comment by roamergu** [comment]
> Sectigo/comodoca is the ssl certificat used by gravatar web site.
> So maybe you have just to disable this feature in Git Extensions as advised in the first messages....

i'ts not true.
the problem be solved in our pc,stefan-wenig is right, before runing git-bash.exe windows will be check signature use ocsp protocol to access oscp.comodoca.com and crl.comodoca.com and so on. when i set timeout to 1 s, everything is ok just delay 1 sec on gpedit
